### PR TITLE
eslint: Enable "semi" rule

### PR DESCRIPTION
**Pull request by suomiy**
- fixes semicolons in all files

after the discussion with @mareklibra in #9466, we concluded it could be useful to have eslint rule for that

### CODE REVIEW

File: pkg/apps/application-list.jsx [hunk 1]
```diff
@@ -69,7 +69,7 @@ class ApplicationRow extends React.Component {
                         {comp.summary}
                         <div className="alert alert-danger alert-dismissable">
                             <button className="close"
-                                    onClick={left_click(() => { this.setState({ error: null }) })}>
+                                    onClick={left_click(() => { this.setState({ error: null }); })}>
```
Comment (martinpitt) [inline pkg/apps/application-list.jsx]: IMHO this is a bit over the top. I think this should rather drop the unnecessary { } *and* the semicolon for a single-line function call. I. e. `() => this.setState(...)`.

File: pkg/storaged/storage-controls.jsx [hunk 2]
```diff
@@ -160,7 +160,7 @@ var StorageBlockNavLink = React.createClass({
         var parts = utils.get_block_link_parts(client, block.path);
 
         var link = (
-            <a role="link" tabIndex="0" onClick={() => { cockpit.location.go(parts.location) }}>
+            <a role="link" tabIndex="0" onClick={() => { cockpit.location.go(parts.location); }}>
```
Comment (martinpitt) [inline pkg/storaged/storage-controls.jsx]: Another such case

File: pkg/storaged/storage-controls.jsx [hunk 3]
```diff
@@ -185,7 +185,7 @@ class StorageOnOff extends React.Component {
             var promise = self.props.onChange(val);
             if (promise) {
                 promise.always(function() {
-                    self.setState({ promise: null })
+                    self.setState({ promise: null });
                 });
```
Comment (martinpitt) [inline pkg/storaged/storage-controls.jsx]: and another one; also, `function() {` → `() =>`

**Comment by suomiy** [comment]
> This makes sense, thank you! I'd just like to avoid some excessive punctuation and braces for single-line arrow functions.

I guess that is a bit too much. Added `"omitLastInOneLineBlock": true` option.

**Comment by suomiy** [comment]
You are right, sorry I forgot to test it. Added corresponding jshint rule.

**Pull request merged**